src/main/kotlin/com/terraformation/backend/accelerator/api/CohortsController.kt

```diff
@@ -0,0 +1,115 @@
+import com.terraformation.backend.db.default_schema.CohortPhase
+import com.terraformation.backend.db.default_schema.ParticipantId
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@AcceleratorEndpoint
+@RequestMapping("/api/v1/accelerator/cohorts")
+@RestController
+class CohortsController(private val cohortStore: CohortStore) {
+  @ApiResponse(responseCode = "200")
+  @ApiResponse404
+  @GetMapping("/{cohortId}")
+  @Operation(summary = "Gets information about a single cohort.")
+  fun getCohort(@PathVariable("cohortId") cohortId: CohortId): CohortResponsePayload {
+    val cohort = cohortStore.fetchOneById(cohortId)
+    return CohortResponsePayload(CohortPayload(cohort))
+  }
+
+  @ApiResponse(responseCode = "200")
+  @GetMapping()
+  @Operation(summary = "Gets the list of cohorts.")
+  fun getCohortList(): CohortListResponsePayload {
+    val cohortList = cohortStore.findAll()
+    return CohortListResponsePayload(cohortList.map { CohortPayload(it) })
+  }
+
+  @ApiResponse(responseCode = "200", description = "The cohort was created successfully.")
+  @PostMapping
+  @Operation(summary = "Creates a new cohort.")
+  fun createBatch(@RequestBody payload: CreateCohortRequestPayload): CohortResponsePayload {
+    val insertedModel = cohortStore.create(payload.toModel())
+    return CohortResponsePayload(CohortPayload(insertedModel))
+  }
+
+  @ApiResponse(responseCode = "200")
+  @ApiResponse404
+  @DeleteMapping("/{cohortId}")
+  @Operation(summary = "Deletes a single cohort.")
+  fun deleteCohort(@PathVariable("cohortId") cohortId: CohortId): SimpleSuccessResponsePayload {
+    cohortStore.delete(cohortId)
+    return SimpleSuccessResponsePayload()
+  }
+
+  @ApiResponse(responseCode = "200")
+  @ApiResponse404
+  @PutMapping("/{cohortId}")
+  @Operation(summary = "Updates the information within a single cohort.")
+  fun updateCohort(
+      @PathVariable("cohortId") cohortId: CohortId,
+      @RequestBody payload: UpdateCohortRequestPayload
+  ): CohortResponsePayload {
+    cohortStore.update(cohortId, payload::applyChanges)
+    return getCohort(cohortId)
+  }
+}
+
+data class CohortPayload(
+    val id: CohortId,
+    val name: String,
+    val participantIds: List<ParticipantId>?,
+    val phase: CohortPhase
+) {
+  constructor(
+      cohort: ExistingCohortModel
+  ) : this(
+      id = cohort.id,
+      name = cohort.name,
+      participantIds = cohort.participantIds,
```

Comment:
Could maybe add `.orNull()` here so the payload field gets set to null if there are no participants. (`orNull()` is our own extension function, not a Kotlin standard library thing.)